paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/SepaMandateActivity.kt

```diff
@@ -0,0 +1,91 @@
+package com.stripe.android.paymentsheet.ui
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Button
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.stripe.android.common.ui.BottomSheet
+import com.stripe.android.common.ui.rememberBottomSheetState
+import com.stripe.android.uicore.StripeTheme
+import com.stripe.android.ui.core.R as StripeUiCoreR
+
+@OptIn(ExperimentalMaterialApi::class)
+internal class SepaMandateActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val merchantName = runCatching {
+            requireNotNull(SepaMandateContract.Args.fromIntent(intent)) {
+                "SepaMandateActivity was started without arguments."
+            }
+        }.getOrNull()?.merchantName
+
+        if (merchantName == null) {
+            finish()
```

Comment:
Should we log a message for easier debugging for integrators?

Reply:
The only time this could happen is if you don't supply args. I can log a message, but this shouldn't ever happen. This is just a safety to prevent the app from crashing when attackers try to launch our activity illegally.